src/etc/inc/plugins.inc.d/ipsec.inc

```diff
@@ -843,820 +812,756 @@ function ipsec_configure_spd()
+        set_single_sysctl('net.key.preferred_oldsa', '-30');
+    } else {
+        set_single_sysctl('net.key.preferred_oldsa', '0');
+    }
 
-                @chmod($publickeypath, 0600);
+    $ipseccfg = $config['ipsec'];
+    $a_phase1 = isset($config['ipsec']['phase1']) ? $config['ipsec']['phase1'] : [];
+    $a_phase2 = isset($config['ipsec']['phase2']) ? $config['ipsec']['phase2'] : [];
+    $a_client = isset($config['ipsec']['client']) ? $config['ipsec']['client'] : [];
 
-                $ph1publickeyfile = "${publickeypath}/publickey-local-{$ph1ent['ikeid']}.pem";
-                if (!file_put_contents($ph1publickeyfile, $keyPair['publicKey'])) {
-                    log_error(sprintf('Error: Cannot write phase1 local public key file for %s', $ph1ent['name']));
-                    @unlink($ph1publickeyfile);
-                    continue;
-                }
-                @chmod($ph1publickeyfile, 0600);
+    if (!isset($ipseccfg['enable'])) {
+        /* try to stop charon */
+        mwexec('/usr/local/etc/rc.d/strongswan stop');
 
-                $ph1privatekeyfile = "${privatekeypath}/privatekey-local-{$ph1ent['ikeid']}.pem";
-                if (!file_put_contents($ph1privatekeyfile, $keyPair['privateKey'])) {
-                    log_error(sprintf('Error: Cannot write phase1 local private key file for %s', $ph1ent['name']));
-                    @unlink($ph1privatekeyfile);
-                    continue;
-                }
-                @chmod($ph1privatekeyfile, 0600);
+        /* wait for process to die */
+        sleep(2);
 
-                $pskconf .= " : " . ipsec_get_key_type($ph1privatekeyfile) . " {$ph1privatekeyfile}\n";
-            }
+        /* disallow IPSEC, it is off */
+        mwexec("/sbin/ifconfig enc0 down");
 
-            if (!empty($ph1ent['peer-kpref'])) {
-                $keyPair = ipsec_lookup_keypair($ph1ent['peer-kpref']);
-                if (!$keyPair || empty($keyPair['publicKey'])) {
-                    log_error(sprintf('Error: Invalid phase1 peer key pair reference for %s', $ph1ent['name']));
-                    continue;
-                }
+        return 0;
+    } else {
+        if (get_single_sysctl('net.inet.ipsec.def_policy') == "") {
+            mwexec("/sbin/kldload ipsec");
+        }
+        mwexec("/sbin/ifconfig enc0 up");
 
-                @chmod($publickeypath, 0600);
+        if ($verbose) {
+            echo 'Configuring IPsec VPN...';
+        }
 
-                $ph1publickeyfile = "${publickeypath}/publickey-peer-{$ph1ent['ikeid']}.pem";
-                if (!file_put_contents($ph1publickeyfile, $keyPair['publicKey'])) {
-                    log_error(sprintf('Error: Cannot write phase1 peer public key file for %s', $ph1ent['name']));
-                    @unlink($ph1publickeyfile);
-                    continue;
+        // cleanup legacy ipsec.conf files
+        if (is_file("/usr/local/etc/ipsec.conf") && is_dir("/usr/local/etc/ipsec.d")) {
```

Comment:
ipsec.conf is possibly sampled by strongswan package so always there even if we delete it (but need to double-check)